scipy/sparse/bsr.py

```diff
@@ -407,14 +407,14 @@ def tobsr(self,blocksize=None,copy=False):
         else:
             return self
 
-    def tocsr(self):
-        return self.tocoo(copy=False).tocsr()
+    def tocsr(self, copy=False):
+        return self.tocoo(copy=copy).tocsr()
```

Comment:
What @pv is saying here is that due to the current implementation of `coo_matrix.tocsr()`, it will always produce a copy no matter what. So doing `self.tocoo(copy=False)` is always acceptable, so long as you then convert to CSC/CSR format.

He also suggests using the copy kwarg in the final `.tocsr()` call, because that will protect this method from causing trouble in the future if the CSR/CSC conversion changes to allow non-copying behavior.

The same goes for the BSR->COO->CSC conversion below.